packages/ove-app-networks/src/client/common/networks.js

```diff
@@ -13,6 +13,235 @@ $(function () {
     });
 });
 
+runOperation = function (message) {
+    // Helper method to retrieve a property from an element
+    const getFromElement = function (e, x) {
+        let m;
+        let n;
+        if (!x || x.indexOf('.') === -1) {
+            n = x;
+        } else {
+            m = x.substring(x.indexOf('.') + 1);
+            n = x.substring(0, x.indexOf('.'));
+        }
+        const p = e[n] || (e.attributes ? e.attributes[n] : undefined);
+        return m ? getFromElement(p, m) : p;
+    };
+
+    // Helper method to retrieve a property from a message
+    const getFromMessage = function (message, p1, p2) {
+        return message[p1] ? message[p1][p2] : undefined;
+    };
+
+    // IMPORTANT: There are no logs within the filter evaluation operations to ensure
+    // the most optimum performance. The logs corresponding to the operation carried out
+    // can be used for debugging purposes.
+    const evaluate = function (element, filter) {
+        // Evaluate as a number
+        const evaluateN = function (element, filter) {
+            return +(evaluate(element, filter));
+        };
+        // Evaluate as a string
+        const evaluateS = function (element, filter) {
```

Comment:
is there performance value in lifting these function definitions out (above) this scope?

Reply:
No, they would work out be the same when compiled. You can try checking by generating a min out of this JS file. The format that we have right now makes it easy to read and follow.